pkg/segment/results/mresults/metricresults.go

```diff
@@ -526,3 +535,18 @@ func (r *MetricsResult) FetchPromqlMetricsForUi(mQuery *structs.MetricsQuery, pq
 
 	return httpResp, nil
 }
+
+func calculateInterval(timerangeSeconds uint32) (uint32, error) {
+	// If timerangeSeconds is greater than 10 years reject the request
+	if timerangeSeconds > 10*365*24*60*60 {
```

Comment:
maybe define this number calc as a CONST : `TEN_YEARS_IN_SECS` and use in both places